### ISSUE TITLE
Study the Stack - I

### ISSUE THREAD

**Issue by OneFabric** [OneFabric/alphaReader#12]
Watch linked YouTube video, and comment below on 5 things you learned based on the video.

Here is a sample template:

```
Point made:

Why I like it:

How does this relate to web app development?:
``` 

Link: https://www.youtube.com/watch?v=ZVnjOPwW4ZA

**Comment by OneFabric** [comment]
I first need a course in react looks like. My fundamentals are not strong in front-end. Starting this course by the same YouTuber.

https://www.youtube.com/watch?v=SqcY0GlETPk&t=0s

**Comment by OneFabric** [comment]
**Point made:** Division of project structure - dev dependencies won't be deployed.

**Why I like it:** I didn't realize that in today's world with a lot of access to modern code development, a separate section can be devoted to developer tooling. Until I saw the relevant explanation in the video, I thought libraries were solely installed to outsource (out-code?) functionality to existing tools for the business need.

**How does this relate to web app development?:** It makes one very specific section _very_ clear - the portion of `package.json` meant for dev dependencies. I never fully understood the configuration files, and understanding one section takes me closer to the broader ecosystem.

Refers to this [YouTube Video](https://www.youtube.com/watch?v=SqcY0GlETPk).

**Comment by OneFabric** [comment]
**Point (topic) made**: DOM, React DOM.

**Why I like it**: While I was familiar with DOM, and believed it to be at the intersection of HTML and JavaScript, I struggled to understand React frameworks for a very, _very_ long time.  I read the tutorial on the react website, but the key insight is how the modularization works. How that modularization works, clicked today.

**How does this relate to web app development?:** If the different resources a web page needs can be modularized so effectively with React, then by thinking about the design ahead, a lot of developer time can be saved. 

For example, if the requirements for a web app points towards a lot of dialog boxes, then the product manager can ask the team to focus on dialog boxes first. Once that's done, the middleware and back-end team can do a check-in to see if their API work is impacted by the dialog box coding.

In other words, the existence of a framework greatly speeds the synchronization of a fast-moving, cross-tech stack team, by allowing developers to prioritize well.

**Comment by OneFabric** [comment]
Closing as not done. See #13 for details.

**Comment by OneFabric** [comment]
**Point made:** React apps use Pascal casing.

**Why I like it:** It's difficult to know the conventions as a newbie, and saves a lot of frustration (and hopefully compiler issues) if I approach naming correctly.

**How does this relate to web app development?:** I am used to camel casing in c#.
